### PR TITLE
Fix building bcopy.c for CHERI

### PR DESCRIPTION
**Pull request by zxombie**
- Fix the location of __capability in pointers
 - Use __cheri_addr when getting an address value

### CODE REVIEW

**Comment by jrtc27** [comment]
What surprises me is we've managed to not have memcpy_c/memmove_c/memset_c on RISC-V, despite the fact that I can trivially make the compiler generate a call to one. I guess we just don't have enough hybrid userspace code, and the hybrid kernel has its own libc which _does_ have memcpy_c? But yes these fixes look correct. We probably want to bring back memcpy_c/memset_c/bcopy_c as MI files for `.if ${MACHINE_CPU:Mcheri}` (and let MIPS continue to override the first two with its own MD implementations if it really wants).

**Comment by zxombie** [comment]
We would also need a MI value to add to CFLAGS for these.

**Comment by jrtc27** [comment]
That should come from bsd.cpu.mk or bsd.compat.mk.

File: lib/libc/string/Makefile.inc [hunk 1]
```diff
@@ -27,6 +27,10 @@ MISRCS+=bcmp.c bcopy.c bzero.c explicit_bzero.c \
 	wmemcmp.c \
 	wmemcpy.c wmemmove.c wmemset.c
 
+.if ${MACHINE_CPU:Mcheri}
+MDSRCS+=memcpy_c.c memmove_c.c memset_c.c
```
Comment (jrtc27) [inline lib/libc/string/Makefile.inc:31]:
```suggestion
.if ${MACHINE_CPU:Mcheri} && !${MACHINE_ABI:Mpurecap}
MISRCS+=memcpy_c.c memmove_c.c memset_c.c
```
I think?

**Pull request merged**